py_cui/control_widgets/slider.py

```diff
@@ -4,12 +4,109 @@
 
 
 class SliderImplementation(py_cui.ui.UIImplementation):
-    pass
+
+    def set_bar_char(self,char):
+        self._bar_char = char
+
+    def update_slider_value(self, direction):
+        """ set the value of the slider - increment decrement
+        """
+        # direction , 1 raise value, -1 lower value
+        self._cur_val += (direction * self._step)
+        if (self._cur_val <= self._min_val):
+            self._cur_val = self._min_val
+        if (self._cur_val >= self._max_val):
+            self._cur_val = self._max_val
+
+        return self._cur_val
+
+    def get_slider_value(self):
+        """return current slider value
+        """
+        return self._cur_val
+
+    def set_slider_step(self,step):
+        """change step value
+        """
+        self._step = step
 
 
 class SliderWidget(py_cui.widgets.Widget, SliderImplementation):
-    pass
+    """Widget for a Slider
+    """
+
+    """
+    Attributes
+    ----------
+    _min_val : int
+        Lowest value of the slider
+    _max_val: int
+        Highest value of the slider
+    _step : int
+        Increment from low to high value
+    _cur_val:
+        Current value of the slider
+
+    """
+    _bar_char = '#'
+
+    def __init__(self, id, title, grid, row, column, row_span, column_span,
+                 padx, pady, logger, min_val, max_val, step, init_val):
+
+        super().__init__(id, title, grid, row, column,
+                         row_span, column_span, padx, pady, logger,
+                         selectable=True)
+
+        self._min_val = min_val
+        self._max_val = max_val
+        self._cur_val = init_val
+        self._step = step
+
+        if self._cur_val < self._min_val or self._cur_val > self._max_val:
+            raise Exception('initial value must be between {} and {}'
```

Comment:
This could also be moved to `SliderImplementation`'s initializer. May also want to change `Exception` to `PyCUIError` from `py_cui.errors`. Not much of a difference, but just for aesthetic reasons.

Reply:
done, and test class updated